### ISSUE TITLE
ffmpeg -encoders

### ISSUE THREAD

**Issue by morgangiraud** [fluent-ffmpeg/node-fluent-ffmpeg#336]
Hello people,
First, the 2.0 release seems pretty nice, congratz.

I just encounter a problem recently on a ubuntu server
.
The ffmpeg command was missing the `-encoders` flag which would cause the whole package to fail on every call since he is checking each time for availableEncoders in `capabilities.js`.

It appends in two different install i've done. One from _the official repo of my ubuntu server_ and one from _Jon Severinsson's FFmpeg PPA_ repo.
I had to rebuild a ffmpeg on my own from the source to ensure this flag was existing in the command.

I don't how to solve this issue, but maybe you shouldn't rely on this flag anymore.

**Comment by hiddenmin** [comment]
+1 for this issue.

I am using ffmpeg from ppa:jon-severinsson/ffmpeg on Ubuntu 12.04

**Comment by njoyard** [comment]
I may add an option for those old versions without `-encoders` to bypass checks. That's really weird, though, this option is as old as time itself AFAIK.

**Comment by vogonistic** [comment]
I have the same issue. There is an option called `-codecs` that provides the same information and more that could be used instead or as a fallback. The flags are slightly different though.

**Comment by njoyard** [comment]
@vogonistic can you tell me which version you're using, and on which OS ?

**Comment by vogonistic** [comment]
If I build ffmpeg myself, I have both `-encoders` and `-codecs`. The problematic Ubuntu version is what I get through circleci.org.

```
ubuntu@box87:~$ ffmpeg
ffmpeg version 0.8.17-4:0.8.17-0ubuntu0.12.04.1, Copyright (c) 2000-2014 the Libav developers
  built on Mar 16 2015 13:26:50 with gcc 4.6.3
The ffmpeg program is only provided for script compatibility and will be removed
in a future release. It has been deprecated in the Libav project to allow for
incompatible command line syntax improvements in its replacement called avconv
(see Changelog for details). Please use avconv instead.
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...
```

Without researching further, I'd say it's not `ffmpeg` at all, just a shim for `avconv` that parses most options like `ffmpeg` with obvious issues.

I think the main issue I have is that `fluent-ffmpeg` fails on the `-encoders` check, but if I run with a simple shim like this it runs just fine, even though that format is different: 

```
#!/bin/bash

if [[ "$@" = "-encoders" ]]; then
  exec ffmpeg -codecs;
fi

exec ffmpeg "$@"
```

**Comment by njoyard** [comment]
Thanks.

For this specific issue I think you're better off filing a bug against avconv and their ffmpeg stub/wrapper.

I'm willing to introduce real avconv support in fluent-ffmpeg but it should be complete and run the actual avconv binary. And that's a lot of work...

**Comment by vogonistic** [comment]
I agree that they've failed in their coverage of `ffmpeg` options, but I don't think it's ok for `fluent-ffmpeg` to fail completely because `-encoders` returns exit code `1` when the actual work I want it to do works fine.

**Comment by ahmed-sharief5** [comment]
s3 bucket signed url is not working its showing 'No such file or directory'
